Alerting docs: update Introduction page (#89155)
grafana/grafana: docs/sources/alerting/fundamentals/_index.md

The following diagram gives you an overview of Grafana Alerting and introduces you to some of the fundamental features that are the principles of how Grafana Alerting works.

<br/>

> {{< figure src="/media/docs/alerting/alerting-configure-notifications-v2.png" max-width="750px" alt="How Alerting works" caption="How Alerting works" >}}

Author: has `alt` text = "How alerting works", should we change this?

> {{< figure src="/media/docs/alerting/alerting-configure-notifications-v2.png" max-width="750px" alt="How Alerting works" caption="How Alerting works" >}}

Reviewer: sorry

## How it works at a glance

- Grafana Alerting periodically queries data sources and evaluates the condition defined in the alert rule
- If the condition is breached, an alert instance fires

> - Firing (and resolved) alert instances are sent for notifications, either directly to a contact point or through notification policies for more flexibility

Author: @brendamuir, see also here the usage of "for notifications" as well as the "either/or"

Reviewer: great

{{< figure src="/static/img/docs/alerting/unified/multi-dimensional-alert.png" caption="Multiple alert instances from a single alert rule" >}}

> [Alert rules are frequently evaluated](ref:alert-rule-evaluation) and the state of their alert instances is updated accordingly. Only alert instances that are in a firing or resolved state are sent for notifications.

Author: This one is tricky. 

This may be interpreted that an alert instance is sent "as a notification". 

But this is not how it works. 

An alert instance will then be included within an alert group, and a notification will be sent for the entire group. 

If you think, `as` is still correct, please lmk and will accept the suggestion.

> [Alert rules are frequently evaluated](ref:alert-rule-evaluation) and the state of their alert instances is updated accordingly. Only alert instances that are in a firing or resolved state are sent for notifications.

Reviewer: how about "in" then?

> [Alert rules are frequently evaluated](ref:alert-rule-evaluation) and the state of their alert instances is updated accordingly. Only alert instances that are in a firing or resolved state are sent for notifications.

Author: fixed by https://github.com/grafana/grafana/pull/89155/commits/e910891e559e5125b479617d839371e98045e7c8

### Contact points

[Contact points](ref:contact-points) determine the notification message and where notifications are sent. For example, you might have a contact point that sends notifications to an email address, to Slack, to an incident management system (IRM) such as Grafana OnCall or Pagerduty, or to a webhook.

> In the alert rule, you can configure the contact point to receive the alert notifications or use notification policies instead.

Author: https://grafana.com/docs/grafana/next/alerting/alerting-rules/create-grafana-managed-rule/#configure-labels-and-notifications

> Configure who receives a notification when an alert rule fires by either choosing Select contact point or Use notification policy.

It's an either or.

> In the alert rule, you can configure the contact point to receive the alert notifications or use notification policies instead.

Reviewer: really?but notification policys are how and contact points are to who - so you need both..right? i mean...you could use one or the other...but in other occasions you'd use both?

> In the alert rule, you can configure the contact point to receive the alert notifications or use notification policies instead.

Author: ![Screenshot 2024-06-13 at 16 57 52](https://github.com/grafana/grafana/assets/825430/29699174-a0c3-43d8-9a8c-d8fee9bee4ca)

Yes, you use always "both" behind the scene. But from the user perspective, they choose either one or the other:

![Screenshot 2024-06-13 at 16 57 22](https://github.com/grafana/grafana/assets/825430/8302f7e7-de6d-4e7a-8215-ead6be41a814)

![Screenshot 2024-06-13 at 16 57 34](https://github.com/grafana/grafana/assets/825430/5fd31c1f-9f10-4f15-8299-6853a7529470)

That is why I want to simplify by saying you either:
- choose a contact point
- use notification policies for handling alerts with greater flexibility

> In the alert rule, you can configure the contact point to receive the alert notifications or use notification policies instead.

Reviewer: thats the simple use case - if theyre using email and the default contact point - but if they want to use policies and choose contact poitns and add contact points - so i think if you want to go this route - you're going to have to introduce simple vs advanced rule creation concepts

> In the alert rule, you can configure the contact point to receive the alert notifications or use notification policies instead.

Author: @brendamuir, this is the intro, so I'd like not to go into details:

> In the alert rule, you can choose a contact point to receive the alert notifications or use notification policies instead.

This sentence tries to connect ContactPoint & NotificationPolicies with the `How it works` diagram, the above the fold image. See full

> ### Contact points
> [Contact points](https://grafana.com/docs/grafana/latest/alerting/fundamentals/notifications/contact-points/) determine the notification message and where notifications are sent. For example, you might have a contact point that sends notifications to an email address, to Slack, to an incident management system (IRM) such as Grafana OnCall or Pagerduty, or to a webhook.
> 
> In the alert rule, you can choose a contact point to receive the alert notifications or use notification policies instead.
> 
> ### Notification policies
> [Notification policies](https://grafana.com/docs/grafana/latest/alerting/fundamentals/notifications/notification-policies/) provide a flexible method to handle alert notifications for larger systems.
> 
> Notification policies routes alerts to contact points via label matching. Each notification policy consists of a set of label matchers (0 or more) that specify which alert instances (identified by their labels) they handle. Notification policies are defined in a tree structure, where the root of the notification policy tree is the Default notification policy, which ensures all alert instances are handled.
> 
> Each notification policy decides where to send the alert (contact point) and when to send the notification (timing options). Additionally, it can [group multiple firing alert instances into a single notification](https://grafana.com/docs/grafana/latest/alerting/fundamentals/notifications/group-alert-notifications/) to reduce alert noise.

That particular sentence is also mentioned on the Alert rule section. If you think it is better to remove it, please, lmk. 

This is the last pending issue in the review 🙏

### Notification policies

> [Notification policies](ref:notification-policies) provide a flexible method to handle alert notifications by routing alerts to contact points via label matching.

Reviewer: ok...but then you'd need to say why it's more flexible

> [Notification policies](ref:notification-policies) provide a flexible method to handle alert notifications by routing alerts to contact points via label matching.

Reviewer: yes thats much better :)

> [Notification policies](ref:notification-policies) provide a flexible method to handle alert notifications by routing alerts to contact points via label matching.

Author: addressed by https://github.com/grafana/grafana/pull/89155/commits/df325aca56ff486e40442f035e0bbe2d41d4c41b

> Each notification policy consists of a set of label matchers (0 or more) that specify which alert instances (identified by their labels) they handle. Notification policies are defined as a tree structure, where the root of the notification policy tree is the **Default notification policy**, which ensures all alerts are handled.

Reviewer: ```suggestion
Each notification policy consists of a set of label matchers (0 or more) that specify which alert instances (identified by their labels) they handle. Notification policies are defined in a tree structure, where the root of the notification policy tree is the **Default notification policy**, which ensures all alert instances are handled.
```

> {{< figure src="/media/docs/alerting/notification-routing.png" max-width="750px" caption="Notification policy routing" >}}

Reviewer: needs alt text

Author: fixed by https://github.com/grafana/grafana/pull/89155/commits/167223d4a517cd81812a4cb5a47511e144bd5b7f

> Each notification policy is responsible for deciding the contact point and when to send the notification. Additionally, it can [group multiple firing alert instances into a single notification](ref:group-alert-notifications) to reduce alert noise.

Reviewer: Each notification policy decides where to send the alert (contact point) and when to send the notification (what is this dependent on?? evaluation/grouping, etc.? we could be more specific about that

Author: ok, some reasoning about this update. I wanted to to keep it short because:

1 - This topic is elaborated in more details at: [Intro > Notifications](https://grafana.com/docs/grafana/next/alerting/fundamentals/notifications/) and [Intro > Notification > Notification Policies](https://grafana.com/docs/grafana/next/alerting/fundamentals/notifications/notification-policies/), so I didn't want to extend much on the first Intro, which was intended to be a bit shorter.

2 - The main CTA or goal of this paragraph is the link to [group multiple firing alert instances into a single notification](https://grafana.com/docs/grafana/next/alerting/fundamentals/notifications/group-alert-notifications/) that explains in detail how it works.

3 - 

> to send the notification (what is this dependent on?? evaluation/grouping, etc.? we could be more specific about that

ok, if it is ok to keep it short and don't go in much details, a suggestion could be:

> Each notification policy is responsible for deciding the contact point and when to send the notification based on the timing options. Additionally, it can [group multiple firing alert instances into a single notification](https://grafana.com/docs/grafana/next/alerting/fundamentals/notifications/group-alert-notifications/) to reduce alert noise.

wdyt?

Reviewer: ok sounds good...but can we have the first part of my suggestion? this bit: Each notification policy decides where to send the alert (contact point) and when to send the notification. Additionally ---

Author: Yes!! didn't know it was a suggestion. Do you want to mention the timing options?

Each notification policy decides where to send the alert (contact point) and when to send the notification. 

vs.

Each notification policy decides where to send the alert (contact point) and when to send the notification (timing options).

Reviewer: second fine with me

Author: addressed by https://github.com/grafana/grafana/pull/89155/commits/0c5207a6185bdd1fa3f3615ff56386070b194e13

> {{< figure src="/media/docs/alerting/alerting-notification-policy-diagram-v5.png" max-width="750px" alt="A diagram of the notification policy component" >}}

Reviewer: alt text :)

Author: `alt="A diagram of the notification policy component"`, do we want to change it?

Reviewer: sorry was looking for it at the beginning:(

### Silences and mute timings

[Silences](ref:silences) and [mute timings](ref:mute-timings) allow you to pause notifications for specific alerts or even entire notification policies. Use a silence to pause notifications on an ad-hoc basis, such as during a maintenance window; and use mute timings to pause notifications at regular intervals, such as evenings and weekends.

### Architecture

Grafana Alerting is built on the Prometheus model of designing alerting systems. Prometheus-based alerting systems have two main components:

- An alert generator that evaluates alert rules and sends firing and resolved alerts to the alert receiver.
- An alert receiver (also known as Alertmanager) that receives the alerts and is responsible for handling them and sending their notifications.

> Grafana Alerting doesn’t use Prometheus as its default alert generator because it needs to work with many other data sources, not just Prometheus.

Reviewer: ```suggestion
Grafana Alerting doesn’t use Prometheus as its default alert generator because it works with many other data sources; not just Prometheus.
```
